csrc/device_lower/pass/expr_sort.cpp

```diff
@@ -1504,7 +1504,12 @@ void ExprSegmentationSorter::sort() {
   // Need this for initialization of the DAG that is processed
   std::unordered_map<Expr*, ExprGroup*> expr2group;
 
-  auto all_exprs = fusion_->exprs();
+  // Not putting the exprs between allKnownVals() and fusion inputs here
+  // because they are computed using the expr evaluator.
+  auto all_exprs = StmtSort::getExprsBetween(
```

Comment:
I understand this is just fine for now, but wonder if this should be done at the very beginning of the lowering. The analyses we do at the beginning of the lowering can also skip all the exprs between the inputs and the known values, right?

Reply:
Yeah, agree. Expr sort is the first pass that returns a vector of exprs. After changing expr sort, all following passes will benefit automatically. However, anything before expr sorting are all just analysis. And seems that there is no easier way than just changing all these analysis manually one by one. And I would prefer to leave this as future work, because fully build this out is a big project (for example, how do you make scheduler support `x + x.transpose(0, 1)` by hoisting `x.tranpose(0, 1)` to ATen and schedule as `x + y` using the transpose scheduler?)

Reply:
Sounds good to me. Maybe put some comments there?

Reply:
added